src/WorkerChannel.ts

```diff
@@ -119,7 +119,18 @@ export class WorkerChannel implements IWorkerChannel {
     public workerInitRequest(requestId: string, msg: rpc.WorkerInitRequest) {
         // Validate version
         const version = process.version;
-        if (!(version.startsWith('v14.') || version.startsWith('v16.'))) {
+        if (
+            (version.startsWith('v17.') || version.startsWith('v15.')) &&
+            process.env.AZURE_FUNCTIONS_ENVIRONMENT == 'Development'
+        ) {
+            const msg =
+                'Node.js version used (' +

```

Comment:
This message is like _really_ long haha. Can we use the message from the other file instead please?

Reply:
Haha I've shortened the message a bit, but I think it's useful to explicitly mention which node versions are supported on this runtime, following the lead of the other error message in this file. I also think it's good to have a somewhat long warning to attract attention. If you feel strongly about this, though, I can just copy the exact message from the other file.

Reply:
> I think it's useful to explicitly mention which node versions are supported on this runtime

The reason I removed the explicit versions in my suggestion was because it's too easy for those to get out of date. If you want to keep the versions, you could address that concern with a `const` used in both the message and the `if` statement, which ensures they stay in sync. Basically, a more complicated version of this:
```typescript
const supportedVersion = '1';
if (version !== supportedVersion) {
    const message = `You gotta be on version ${supportedVersion}`;
}
```

> I also think it's good to have a somewhat long warning to attract attention.

At least in core tools, it'll show up bright yellow which serves a similar purpose. Subjectively, I would argue users are more likely to skip long messages because they (me too) are lazy. Objectively, we should keep messages short to keep the logs (as a whole) more manageable. This message likely feels more important because of recency bias, but ultimately it's no more or less important than existing logs/warnings. I wouldn't block a PR on this, but I would likely continue to comment if I see long messages